### PR TITLE
Introduce utility class to manage files with the NIO2 API

### PR DESCRIPTION
**Pull request by nagyo**

### CODE REVIEW

**Review by cmark — APPROVED** [review]
🥓

File: commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	 * Delete all files within the specified directory except the exclusions specified. If there are none, the directory is deleted as well.
+	 * The exclusions must be declared as full paths.
+	 */
+	public static void deleteDirectory(final Path directory, Collection<String> exclusions) throws IOException {
+
+		checkArgument(Files.isDirectory(directory), "The specified path is not a directory");
+
+		final List<Path> filteredPaths = Files.walk(directory)
+				.sorted(Comparator.reverseOrder())
+				.filter(path -> exclusions.stream().noneMatch(p -> p.equals(path.toString())))
```
Comment (apeteri) [inline commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java]: If `exclusions` held a `Collection` of `Path` instances, the check would honor file system specific conditions, eg. case sensitivity, according to `Path#equals`'s javadoc. Which may or may not be a good thing. 😄

File: commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+				.collect(toList());
+
+		for (final Path path : filteredPaths) {
+
+			// do not try to delete the containing directory when there are excluded files present inside
+			if (path.equals(directory) && Files.list(directory).findFirst().isPresent()) {
```
Comment (apeteri) [inline commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java]: I think this should be checked for any directory, not just the initial one. Or do we expect the sub-directory itself to show up in `exclusions` in these cases?

**Review by apeteri — APPROVED** [review]
🍃 with a small note.

File: commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java (hunk 1 above)
Comment (nagyo) [reply commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java]: That's a good idea, will change it accordingly!

File: commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java (hunk 2 above)
Comment (nagyo) [reply commons/com.b2international.commons/src/com/b2international/commons/io/PathUtils.java]: Good point, fixing it.

**Pull request merged**